src/v/coproc/tests/two_way_split_copro.h

```diff
@@ -15,39 +15,52 @@
 #include "storage/record_batch_builder.h"
 #include "storage/tests/utils/disk_log_builder.h"
 
+#include <seastar/core/circular_buffer.hh>
+
 // Silly logic, checks if the key_size is evenly divisible by two, places
 // the record in the even or 'odd' collection based on this predicate
 struct two_way_split_copro : public coprocessor {
     two_way_split_copro(coproc::script_id sid, input_set input)
       : coprocessor(sid, std::move(input)) {}
 
-    coprocessor::result apply(
+    ss::future<coprocessor::result> apply(
       const model::topic&,
-      const std::vector<model::record_batch>& batches) override {
-        model::topic even("even");
-        model::topic odd("odd");
+      ss::circular_buffer<model::record_batch>&& batches) override {
+        static const model::topic even("even");
+        static const model::topic odd("odd");
         coprocessor::result r;
-        r.emplace(even, std::vector<model::record_batch>());
-        r.emplace(odd, std::vector<model::record_batch>());
-        for (auto& record_batch : batches) {
-            storage::record_batch_builder even_rbb(
-              raft::data_batch_type, model::offset(0));
-            storage::record_batch_builder odd_rbb(
-              raft::data_batch_type, model::offset(0));
-            record_batch.for_each_record(
-              [&even_rbb, &odd_rbb](model::record&& record) {
-                  if (record.key_size() % 2 == 0) {
-                      even_rbb.add_raw_kv(
-                        record.share_key(), record.share_value());
-                  } else {
-                      odd_rbb.add_raw_kv(
-                        record.share_key(), record.share_value());
-                  }
-              });
-            r[even].emplace_back(std::move(even_rbb).build());
-            r[odd].emplace_back(std::move(odd_rbb).build());
-        }
-        return r;
+        r.emplace(even, ss::circular_buffer<model::record_batch>());
+        r.emplace(odd, ss::circular_buffer<model::record_batch>());
+        return ss::do_with(
+          std::move(r),
+          std::move(batches),
+          [](
+            coprocessor::result& r,
+            ss::circular_buffer<model::record_batch>& batches) {
+              return ss::do_for_each(
+                       batches,
+                       [&r](model::record_batch& record_batch) {
+                           storage::record_batch_builder even_rbb(
+                             raft::data_batch_type, model::offset(0));
+                           storage::record_batch_builder odd_rbb(
+                             raft::data_batch_type, model::offset(0));
```

Comment:
can this be right? This implies a 0 offset every time apply() is called. Maybe this only currently works if in the particular unit tests where this is called, apply is only being called once

Reply:
Offset management isn't even needed, redpanda will write correct offsets, this can be removed